picard/acoustid/manager.py

```diff
@@ -112,13 +130,9 @@ def __fingerprint_submission_finished(self, submissions, document, http, error):
                 timeout=3000
             )
         else:
-            log.debug('AcoustID: successfully submitted')
-            self.tagger.window.set_statusbar_message(
-                N_('AcoustIDs successfully submitted.'),
-                echo=None,
-                timeout=3000
-            )
+            log.debug('AcoustID: %d fingerprints successfully submitted', len(submissions))
             for file, submission in submissions:
                 submission.orig_recordingid = submission.recordingid
                 file.update()
             self._check_unsubmitted()
+        next()
```

Comment:
In case of error during the submission of one batch, it will try to submit next batch, right?
What about removing stuff from `submissions` only on success and try again few times?

Reply:
Fingerprints are only marked as submitted on success. The user can submit again if there are issues. But most likely those issues are permanent.

Reply:
I understand the user can retry manually, what I mean is we could retry few times in case of temporary errors (ie. network) but since items to be submitted are removed from the list before success/failure is checked, it's not possible.
My idea was to invert the logic: try to submit, and on success, remove from `submissions` list. On failure, either try again after a small delay, or just give up immediatly (removing from `submissions` too = current behavior).

For this, you need to pass the whole `submissions` list, and call next with list without batch successfully submitted or call next with current batch again with a small delay/retry.
That's more a change to the code logic, with the idea of implementing delay/retries if needed later.